merk/src/proofs/query/verify.rs

```diff
@@ -37,266 +42,401 @@ pub fn verify(bytes: &[u8], expected_hash: MerkHash) -> CostResult<Map, Error> {
     })
 }
 
-#[cfg(any(feature = "full", feature = "verify"))]
-/// Verifies the encoded proof with the given query
-///
-/// Every key in `keys` is checked to either have a key/value pair in the proof,
-/// or to have its absence in the tree proven.
-///
-/// Returns `Err` if the proof is invalid, or a list of proven values associated
-/// with `keys`. For example, if `keys` contains keys `A` and `B`, the returned
-/// list will contain 2 elements, the value of `A` and the value of `B`. Keys
-/// proven to be absent in the tree will have an entry of `None`, keys that have
-/// a proven value will have an entry of `Some(value)`.
-pub fn execute_proof(
-    bytes: &[u8],
-    query: &Query,
-    limit: Option<u16>,
-    offset: Option<u16>,
-    left_to_right: bool,
-) -> CostResult<(MerkHash, ProofVerificationResult), Error> {
-    let mut cost = OperationCost::default();
-
-    let mut output = Vec::with_capacity(query.len());
-    let mut last_push = None;
-    let mut query = query.directional_iter(left_to_right).peekable();
-    let mut in_range = false;
-    let mut current_limit = limit;
-    let mut current_offset = offset;
+#[derive(Copy, Clone, Debug)]
+pub struct VerifyOptions {
+    pub absence_proofs_for_non_existing_searched_keys: bool,
+    pub verify_proof_succinctness: bool,
```

Comment:
Same for this `verify_proof_succinctness`. (The other two are explicit)

Reply:
Done